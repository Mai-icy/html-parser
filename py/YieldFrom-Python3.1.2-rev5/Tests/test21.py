#
#  Test next and return with value
#

def f(r):
	gi = g(r)
	next(gi)
	try:
		print("f resuming g")
		next(gi)
		print("f SHOULD NOT BE HERE")
	except StopIteration as e:
		print("f caught", repr(e))

def g(r):
	print("g starting")
	yield
	print("g returning", r)
	return r

f(None)
f(42)
