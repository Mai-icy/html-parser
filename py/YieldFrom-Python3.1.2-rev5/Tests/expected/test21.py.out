g starting
f resuming g
g returning None
f caught StopIteration()
g starting
f resuming g
g returning 42
f caught StopIteration(42,)
